markdown/core.py

```diff
@@ -85,7 +86,11 @@ class Markdown:
     callable which accepts an [`Element`][xml.etree.ElementTree.Element] and returns a `str`.
     """
 
-    def __init__(self, **kwargs):
+    toc_tokens: list[TocToken]
+    toc: str
+    Meta: dict[str, Any]
```

Comment:
Why are these here? They only exist via extensions. The core knows nothing of these extensions and should under no circumstances define these.

This is important because third party extensions might want to do something similar and it it not reasonable for us to provide the same for them. If this is a problem, then we need a different solution. This was discussed in depth elsewhere (can't seem to find it now).

Reply:
> The type checking solutions have definitively decided to not support the use case of setting random unrecognized attributes on objects.
>
> ...
>
> We should just say that non-core extensions should not set new attributes on Markdown.

This is a problem because I expect to be able to do this. If there is no workaround, then we might as well forgot about ever completing this.

I found the previous discssion on this in #1383. Note in particular my comments about setters and getters. Perhaps this discussion should be continued there.